stores/shoppingCart.ts

```diff
@@ -16,27 +17,18 @@ export const useShoppingCartStore = defineStore('shoppingCart', {
     items: items(),
     itemToBuy: undefined,
   }),
-  getters: {
-    getItems: () => items(),
-    getItemsByPrefix: () => (prefix: string) =>
-      items().filter((item) => item.urlPrefix === prefix),
-    getItem: () => (id: ID) => items().find((item) => item.id === id),
-    isItemInCart: () => (id: ID) =>
-      items().find((item) => item.id === id) !== undefined,
-    getItemToBuy: ({ itemToBuy }) => itemToBuy,
-  },
-
   actions: {
+    getItemsByPrefix(prefix: string) {
+      return this.items.filter((item) => item.urlPrefix === prefix)
+    },
+    isItemInCart(id: ID) {
+      return this.items.find((item) => item.id === id) !== undefined
+    },
     setItem(payload: ShoppingCartItem) {
-      const existinItemIndex = items().findIndex(
-        (item) => item.id === payload.id
-      )
-      if (existinItemIndex === -1) {
-        localStorage.value = [...localStorage.value, payload]
-      } else {
-        const items = [...localStorage.value]
-        items[existinItemIndex] = payload
-        localStorage.value = items
```

Comment:
setitem is used both to set new item and to replace/update an existing one
why did you delete the 'update/replace/ part?

Reply:
Yes that is correct, but if you follow the usecase of setItem now, the update part is not anymore being used or required